src/app/static/src/app/store/load/actions.ts

```diff
@@ -74,8 +78,11 @@ export const actions: ActionTree<LoadState, RootState> & LoadActions = {
 
         if (!rootGetters.isGuest) {
             await (dispatch("projects/createProject", {name: state.projectName || null}, {root: true}));
-            savedState.projects.currentProject = rootState.projects.currentProject;
-            savedState.projects.currentVersion = rootState.projects.currentVersion;
+            savedState.projects = {
+                currentProject: rootState.projects.currentProject,
+                currentVersion: rootState.projects.currentVersion,
+                previousProjects: []
+            } as any
```

Comment:
not 100% sure but could this be defined in any situation? in which case maybe you just want to check first if it is undefined before resetting it completely

Reply:
I don't think it should ever be defined but I guess that could change in the future and should be better to be safe. I've updated that now to add a guard.

Reply:
I made that change that in that's causing errors now. This should never be set. It is reading the `savedState` variable from the file and we know for json and zip it is not set. And there is a test which saves a file with projects as an empty object and then reads that in using `setFiles` action and tests that it has the currentProject, currentVersion and previousProjects set so I'm going to put this back in to make that test pass. I think not too big of a deal though as we're going to rip this all out shortly